src/main/java/io/anserini/index/IndexCollection.java

```diff
@@ -536,7 +537,10 @@ public void run() throws IOException, InterruptedException {
     // Do a final commit
     if (args.solr) {
       try {
-        solrPool.borrowObject().commit(args.solrIndex);
+        SolrClient client = solrPool.borrowObject();
+        client.commit(args.solrIndex);
```

Comment:
@tokee Is this any different from calling `solrPool.borrowObject().commit(args.solrIndex, false, false)`, where the two `false` are the `waitFlush` and `waitSearcher` parameters?

Reply:
The point was to let the `solrPool` get the `SolrClient` back after use. If it does not, the pool treats the `SolrClient` as a still-borrowed object and refuses to close until the timeout of 1 minute has been reached.

Whether or not the `SolrClient` should wait for the Solr-commit to finish is independent of this. I have no suggestions there: It all depends on whether the user expects to see all the changes visible immediately after Anserini has finished or if it is fine with the given `commitWithin`-timeframe.

Reply:
Ahh, gotcha - I missed the point of the change.